src/utils/flow.js

```diff
@@ -29,3 +31,66 @@ export function unexpectedCase(notValid: empty): void {
 export function immutableUpdate<T: Object>(object: T, ...rest: Object[]): T {
   return Object.assign({}, object, ...rest);
 }
+
+/**
+ * This function takes a string and returns either a valid TabSlug or null, this doesn't
+ * throw an error so that any arbitrary string can be converted, e.g. from a URL.
+ */
+export function toValidTabSlug(tabSlug: any): TabSlug | null {
```

Comment:
nit: please take a string as parameter and do just like in `convertToTransformType` :)

Reply:
There's a `toValidTabSlug` in url-handling, do you think you can make it use this new function?

Reply:
I'd prefer `any`, as then I can take null types and whatever I throw at it. I believe I pass potentially null values as it is.

Reply:
Thanks for catching the `toValidTabSlug` duplication.

Reply:
With "any" Flow won't tell you if you try to pass, I don't know, an object, or a number, which could indicate a bug. 
Currently it's `string | void` (not null).
I won't fight for this but I still think this would be better.